index.js

```diff
@@ -138,3 +138,41 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+//Deliv 1 & 2
+
+const dataIdLi = document.querySelector('ul#animals')
+
+dataIdLi.addEventListener('click', function(event){
+    let sighting = event.target.closest('li')
+    if (event.target.matches('button.like-button')){
+        const likesPtag = event.target.previousElementSibling
+        const likes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${likes} Likes`
+    } else if (event.target.matches('button.delete-button')){
+        sighting.remove()
+    } else if (event.target.matches('button.toggle-update-form-button')){
+       let updateForm = event.target.nextElementSibling
+        if (updateForm.style.display === 'none'){
+            updateForm.style.display = 'block'
+        } else {
+            updateForm.style.display = 'none'
+        }
+    }
+})
+
+dataIdLi.addEventListener('submit', function(event){
+    event.preventDefault()
+    let p = dataIdLi.querySelector('p')
+    let newDesc = event.target[0].value 
+
+    p.textContent = newDesc
+
+
+
+    console.log(newDesc)
+})
```

Comment:
Great work here as well. You are listening for any form submission that occurs within the animals UL element. Then you make sure to prevent the default action of the form. Then you find the p tag and update it's value based on the form data. Don't forget to change the form field value to the new description. Again, we can break it apart into 2 steps.

```
/***** BONUS DELIVERABLE part 1 *****/

animalsUL.addEventListener("submit", function(event){
    event.preventDefault()
    if (event.target.matches('.update-form')){
        handleUpdateFormSubmission(event)
    }
})

/***** BONUS DELIVERABLE part 2 *****/
function handleUpdateFormSubmission(e){
    e.preventDefault()
    const description = e.target.children[0].value
    const animalDescriptionPTag = e.target.parentElement.children[0]
    animalDescriptionPTag.innerText = description
    e.target.value = description
}
```